### PR TITLE
API refactoring

### LINKED ISSUE #114: Refactor application into subsystems

### PR DESCRIPTION
**Pull request by tobloef**
Did some refactoring to the API controller, so that we might close #114 with good conscience.

I thought about splitting the controller, but honestly, why would we? It's so short, it would just ake things more confusing.

We now use a new version of the `Json` method, that utilizes the well known Newtonsoft.JSON package, so that we may specify JSON property keys in our DTOs.

### CODE REVIEW

**Comment by Hjaltesorgenfrei** [comment]
Have you run the Api Simulator against it offline?

**Comment by tobloef** [comment]
No, I couldn't get it to work. I did test it manually with CURL tho, both post and get requests. Can you maybe try the simulator?

**Comment by jlndk** [comment]
closes #70

**Comment by Hjaltesorgenfrei** [comment]
I have run the sim now.
It gives the following errors, followed by every tweet failing:
```
python minitwit_simulator.py http://localhost:80/api
2020-03-26 21:40:58,http://localhost:80/api,3,404,tweet
2020-03-26 21:40:58,http://localhost:80/api,5,404,tweet
2020-03-26 21:40:58,http://localhost:80/api,6,404,tweet
2020-03-26 21:40:59,http://localhost:80/api,9,404,tweet
2020-03-26 21:40:59,http://localhost:80/api,10,404,tweet
2020-03-26 21:40:59,http://localhost:80/api,15,404,tweet
2020-03-26 21:41:00,http://localhost:80/api,16,404,tweet
2020-03-26 21:41:00,http://localhost:80/api,17,404,tweet
2020-03-26 21:41:00,http://localhost:80/api,18,404,tweet
2020-03-26 21:41:00,http://localhost:80/api,20,404,tweet
```

File: MiniTwit.Web.App/Controllers/ApiController.cs [hunk 1]
```diff
@@ -34,233 +35,215 @@ IUserRepository userRepository
...
+                .Select(GetMessageDTO.FromMessage);
             return Json(messages);
         }
         
 
         [Route("[controller]/msgs/{username}")]
         [HttpPost]
         [AllowAnonymous]
-        public async Task<IActionResult> UserPostMessages(string username,
-            [FromQuery(Name = "latest")] int? latestMessage, [FromBody] MessagePost messagePost)
```
Comment (Hjaltesorgenfrei) [inline MiniTwit.Web.App/Controllers/ApiController.cs:126]: As a minimum this does not work. Dont merge until it has run for 2000 messages with the sim locally.

**Review by Hjaltesorgenfrei — CHANGES_REQUESTED** [review]

File: MiniTwit.Web.App/Controllers/ApiController.cs (hunk 1 above)
Comment (Hjaltesorgenfrei) [reply MiniTwit.Web.App/Controllers/ApiController.cs:126]: I think the follow post action is also broken.

**Comment by Hjaltesorgenfrei** [comment]
> closes #70
This issue is in reference to the Repository dtos, which it does not fix.

**Pull request closed without merging**